src/strategies/soundxyz-metadata/extractor.mjs

```diff
@@ -0,0 +1,113 @@
+// @format
+import { env } from "process";
+import { encodeCallSignature, decodeCallOutput } from "eth-fun";
+
+export const name = "soundxyz-metadata";
+export const props = {
+  signatures: {
+    tokenToEdition: "tokenToEdition(uint256)",
+    editions: "editions(uint256)",
+  },
+  contract: {
+    address: "0x01ab7d30525e4f3010af27a003180463a6c811a6",
+  },
+  options: {
+    url: env.RPC_HTTP_HOST,
+    headers: {
+      Authorization: `Bearer ${env.RPC_API_KEY}`,
+    },
+  },
+};
+
+export function init(tokenId) {
+  const data = encodeCallSignature(
+    props.signatures.tokenToEdition,
+    ["uint256"],
+    [tokenId]
+  );
+
+  const from = null;
+  return {
+    messages: [
+      {
+        type: "json-rpc",
+        commissioner: name,
+        options: props.options,
+        version: "0.0.1",
+        method: "eth_call",
+        params: [
+          {
+            from,
+            to: props.contract.address,
+            data,
+          },
+          "latest",
+        ],
+        results: null,
+        error: null,
+      },
+    ],
+  };
+}
+
+export function update(message) {
+  // because "editions" returns a struct with 9 fields
+  if (message.results.length === 578) {
+    const editionMetadata = decodeCallOutput(
+      [
+        "address",
+        "uint256",
+        "uint32",
+        "uint32",
+        "uint32",
+        "uint32",
+        "uint32",
+        "uint32",
+        "address",
+      ],
+      message.results
+    );
+
+    return {
+      messages: [],
+      write: JSON.stringify(editionMetadata),
```

Comment:
This is great, but it'd be even better if all those variables currently represented in a list were named fields. So can we please returned a stringifyed object with property names here?

Reply:
To return a stringifyed object I have added a transformer. Like the existing soundxyz strategy it reads the line, decodes the output and transform it into an object before writing it to disc.

Note: Since the implementation is like the existing soundxyz it is to prone its problems also (duplicate lines).